plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShift3ResourceUtils.java

```diff
@@ -0,0 +1,80 @@
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.requirement;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+
+import org.eclipse.osgi.util.NLS;
+import org.jboss.reddeer.common.logging.Logger;
+import org.jboss.tools.openshift.common.core.connection.ConnectionURL;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.reddeer.exception.OpenShiftToolsException;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+
+/**
+ * @author adietish@redhat.com
+ */
+public class OpenShift3ResourceUtils {
+	
+	private static Logger LOGGER = new Logger(OpenShift3ResourceUtils.class);
+
+	public static Connection getConnection(String connectionUrlString) {
+		try {
+			ConnectionURL connectionUrl = ConnectionURL.forURL(connectionUrlString);
+			Connection connection = ConnectionsRegistrySingleton.getInstance().getByUrl(connectionUrl, Connection.class);
+			assertNotNull(NLS.bind("No connection for {0} exists", connectionUrl.toString()), connection);
+			return connection;
+		} catch (UnsupportedEncodingException | MalformedURLException e) {
+			throw new OpenShiftToolsException(NLS.bind("Could not lookup connection {0}: {1}", connectionUrlString, e));
+		}
+	}
+
+	/**
+	 * Returns the project with the given name that exists on the server that the given connection points to. Returns {@code null} if none was found.
+	 * 
+	 * @param name
+	 * @param connection
+	 * @return
+	 */
+	public static IProject getProject(String name, Connection connection) {
+		assertNotNull(connection);
+
+		if (StringUtils.isEmpty(name)) {
+			return null;
+		}
+		
+		for (IResource resource : connection.getResources(ResourceKind.PROJECT)) {
+			if (name.equals(resource.getName())) {
+				return (IProject) resource;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Returns {@code true} if the given connection has a project with the given name.
+	 * 
+	 * @param name the name of the project that is looked for
+	 * @param connection the connection to use to look up the project
+	 * @return
+	 */
+	public static boolean hasProject(String name, Connection connection) {
+		return getProject(name, connection) != null;
```

Comment:
Not directly, but nested. The request in execute method throws the exception "Unable to execute request to...". But maybe it's really rare case and we dont have to bother with it.